src/incoming_msg.cc

```diff
@@ -14,39 +15,37 @@ IncomingMsg::~IncomingMsg() {
 }
 
 Napi::Value IncomingMsg::IntoBuffer(const Napi::Env& env) {
-#if !(NODE_RUNTIME_ELECTRON && NODE_MODULE_VERSION >= 109)  // 109 is Electron v21 and up
-    if (moved) {
-        /* If ownership has been transferred, do not attempt to read the buffer
-           again in any case. This should not happen of course. */
-        ErrnoException(env, EINVAL).ThrowAsJavaScriptException();
-        return env.Undefined();
+    if (!hasElectronMemoryCage(env)) {
```

Comment:
It might be faster to make this variable static so that it is calculated only once.

```suggestion
	static const auto noElectronMemoryCage = !hasElectronMemoryCage(env);
    if (noElectronMemoryCage) {
```

Reply:
I am actually not sure, because the class/function is being constructed and destroyed for each incoming message. So for each message then the function is called and re-evaluated. 

That's why I opted for caching in the function one level up...

What's your thought about this?

Reply:
Yeah, it makes sense to reuse the cache for all the messages. What about making the check a static property of IncomingMsg?

Reply:
Yes, good idea! I can make the changes on Thursday morning!

Reply:
It can be a separate PR. This should work for now.

Reply:
I think I have it, testing as we speak. Will commit in this branch.

Reply:
Wasn't able to make it a class member (at this moment), as I need the passed on `env`, which is only passed in the `IntoBuffer` function. At least one function call is reduced.